src/main/java/se/iths/imagestorage/repository/FileSystemRepository.java

```diff
@@ -0,0 +1,41 @@
+package se.iths.imagestorage.repository;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.multipart.MultipartFile;
+import se.iths.imagestorage.entity.Image;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Repository
+public class FileSystemRepository {
+    private final Logger log = LoggerFactory.getLogger(FileSystemRepository.class);
+
+    public void uploadImage(MultipartFile file, Image image){
+        Path path = Paths.get(image.getPath());
+
+        try {
+            Files.createDirectories(path.getParent());
+            byte[] bytes = file.getBytes();
+            Files.write(path, bytes);
+        } catch (IOException e) {
+            log.error("Error while creating a directory. Make sure that the file name doesn't exist. Error message: "  + e);
```

Comment:
This logs the error in the spring boot container as an "error" message. Though, it's not only for creating directories. All three of these code lines can throw an error, that's why I made it more generic in my example. We could have a separate try/catch for just the createDirectories if we prefer to post a more specific error.

Also using formatted text ("text {0}", e) instead of concatenation is usually better.